src/actions/guides/frontend/html_forms.cr

```diff
@@ -0,0 +1,512 @@
+    </textarea>
+    ```
+
+    ## Select fields
+
+    For select fields, you'll use a combination of `select_tag` and the `options_for_select` methods.
+    The selected value will be determined by the current value of the attribute (i.e. `op.car_make`)
+
+    ```crystal
+    select_tag(op.car_make, class: "custom-select") do
+      options_for_select(op.car_make, [{"Honda", 1}, {"Toyota", 2}])
+    end
+    ```
+
+    ```html
+    <select name="param_key:car_make" class="custom-select">
+      <option value="1">Honda</option>
+      <option value="2" selected="true">Toyota</option>
+      <option value="3">Ford</option>
+      <option value="4">Volkswagen</option>
+    </select>
+    ```
+
+    The second argument to `options_for_select` takes an `Array(Tuple(String, String | Int32 | Int64))`. If your data is coming from a query, you can easily map that data in to this format.
+
+    ```crystal
+    # Use with `options_for_select(op.car_make, options_for_cars)`
+    private def options_for_cars
+      CarQuery.all.map do |car|
+        {car.make, car.id}
+      end
+    end
+    ```
+
+    ## Checkboxes / Radios
+
+    ### Checkbox
+
+    The `checkbox` method will auto generate a secondary hidden input that will hold the
+    unchecked value of your attribute. This allows params to still contain a value even if a
+    checkbox is not checked.
+
+    ```crystal
+    checkbox(op.with_cheese, "no", "yes", class: "custom-check")
+    ```
+
+    ```html
+    <input type="hidden" name="param_key:with_cheese" value="no" />
+    <input type="checkbox"
+           id="param_key_with_cheese"
+           name="param_key:with_cheese"
+           value="yes"
+           class="custom-check"
+           checked="true" />
+    ```
+
+    ### Radio
+
+    ```crystal
+    input(op.question_five, type: "radio", name: "\#{op.param_key}:\#{op.question_five.name}", value: "Yes")
+    input(op.question_five, type: "radio", name: "\#{op.param_key}:\#{op.question_five.name}", value: "No")
+    ```
+
+    ```html
+    <input type="radio"
+           name="param_key:question_five"
+           value="Yes" />
+    <input type="radio"
+           name="param_key:question_five"
+           value="No" />
+    ```
+
+    > ISSUE REF: https://github.com/luckyframework/lucky/issues/1023
+
+    ## Buttons
+
+    ### button tag
+
+    ```crystal
+    button("Go!", role: "submit", class: "btn")
+    ```
+
+    ```html
+    <button role="submit" class="btn">Go!</button>
+    ```
+
+    ### submit input
+
+    ```crystal
+    submit("Go!", class: "btn")
+    ```
+
+    ```html
+    <input type="submit" value="Go!" class="btn" />
+    ```
+
+    ## Labels
+
+    ```crystal
+    label_for(op.first_name)
+
+    # or with custom text
+
+    label_for(op.first_name, "Enter your First name:")
+
+    # or using a block
+
+    label_for(op.first_name) do
+      strong("First Name: ")
+    end
+    ```
```

Comment:
This looks funny when rendered so I think maybe each example should be split into its own code block

<img width="880" alt="Screen Shot 2020-03-05 at 3 53 44 PM" src="https://user-images.githubusercontent.com/22394/76024716-95202180-5ef9-11ea-9e17-c6454191dd8b.png">

Reply:
That's weird. Why are the comments so faded on your screen?

<img width="583" alt="Screen Shot 2020-03-05 at 1 07 52 PM" src="https://user-images.githubusercontent.com/2391/76025706-592d9200-5ee2-11ea-9bb0-239c58856f0d.png">

Reply:
but yeah, I'll break them out in to separate blocks... but I wonder if that is another weird css bug?